### DISCUSSION TITLE
Pipedream and Deepgram Audio File transcription

### DISCUSSION THREAD

**Discussion by avgindguy** [deepgram/community#199] [General help] [answered]
Followed the process from a blog post I found on how to use Python code in a Pipedream flow to automatically transcribe audio files.  in my case though it does not work.  Plus I see now in Pipedream there is an action now for transcription, though it appears that will only work with a URL string, which is not what I am looking for.  I need to retrieve an audio file from my Google Drive and then have deepgram transcribe it.  My trigger in Pipedream works, I also download the file to a tmp directory in Pipedream and that works.  My ultimate use case is to grab a recording between a contact center agent and a customer, have it transcribed with speaker separation (like the Podcast transcription with deepgram that works very well), and then have the transcription summarized and analyzed by AI.  

Appreciate any advice, still learning 

**this code works for podcast in my pipedream flow**

```
import requests
--
def   handler(pd: "pipedream"):
url = pd.steps["trigger"]["event"]["enclosures"][0]["url"]
token = token =   pd.inputs["deepgram"]["$auth"]['api_key']
listen = "https://api.deepgram.com/v1/listen?tier=enhanced&punctuate=true&diarize=true&paragraphs=true"
headers = { "Authorization": f"Token {token}" }
json = { "url": url }
r =   requests.post(listen, json=json, headers=headers)
response =   r.json()
transcript =   response['results']['channels'][0]['alternatives'][0]['paragraphs']['transcript']
return transcript

**this code I found on the Blog doesnt work in Pipedream to use an audio file from Google**

const fs = require('fs')
const { Deepgram } = require('@deepgram/sdk')

module.exports = defineComponent({
  props: {
    deepgram: {
      type: "app",
      app: "deepgram",
    }
  },
  async run({ steps }) {
      const { name, mimeType } = steps.trigger.event
      const buffer = await fs.promises.readFile(`/tmp/${name}`)
      const source = { buffer, mimetype: mimeType }

      const deepgram = new Deepgram(this.deepgram.$auth.api_key)
     const { results } = await deepgram.transcription.preRecorded(source, { punctuate: true })
     return results.channels[0].alternatives[0].transcript
  }
})
```

**Comment by jjmaldonis** [comment, marked as answer]
Hey @avgindguy! Google Drive handles links to files very strangely. It's difficult to link directly to an image, mp3, or video file, so I'm guessing that's why the code is working for podcasts but not for Google Drive files.

This is how you can get a direct link to an audio file in Google Drive:
First, make sure the file has been shared appropriately. Then copy the link of the audio file you want to transcribe. The URL will look like this: https://drive.google.com/file/d/1nxPDn7MwdepnH2KnGQooTmfBhCQTavwg/view?usp=drive_link

Copy the `fileId` from this URL, which in this case is `1nxPDn7MwdepnH2KnGQooTmfBhCQTavwg`.

Paste the `fileId` at the end of this URL: https://drive.google.com/uc?export=view&id=

The final URL looks like this: https://drive.google.com/uc?export=view&id=1nxPDn7MwdepnH2KnGQooTmfBhCQTavwg and this URL can be sent to Deepgram for transcription. Note that this URL will redirect to a much longer URL when you navigate to the link.

Below is a sample code snippet that shows this working. I didn't try this with Pipedream since I haven't set up an account with them, but if Pipedream is working with podcasts then it should also work with this kind of Google Drive URL.

```python
from deepgram import Deepgram
import json
import os


DEEPGRAM_API_KEY = os.environ["DEEPGRAM_API_KEY"]  # Your Deepgram API Key
deepgram = Deepgram(DEEPGRAM_API_KEY)


def transcribe_audio(get_url: str):
    options = {
        "smart_format": True,
        "model": "nova",
    }
    source = {"url": get_url}
    r = deepgram.transcription.sync_prerecorded(source, options)
    print(json.dumps(r, indent=2))


if __name__ == "__main__":
    transcribe_audio(get_url="https://drive.google.com/uc?export=view&id=1nxPDn7MwdepnH2KnGQooTmfBhCQTavwg")
```
